teachertool/src/components/CatalogOverlay.tsx

```diff
@@ -106,34 +127,72 @@ const CatalogList: React.FC = () => {
         announceToScreenReader(lf("Added '{0}' to checklist.", getReadableCriteriaTemplate(c)));
     }
 
+    function getItemIdForTag(tag: string) {
+        return `accordion-item-${tag}`;
+    }
+
+    function onTagExpandToggled(tag: string, expanded: boolean) {
+        if (expanded) {
+            /* await */ addExandedCatalogTagAsync(tag);
+        } else {
+            /* await */ removeExpandedCatalogTagAsync(tag);
+        }
+    }
+
+    const tags = Object.keys(criteriaGroupedByTag);
+    if (tags.length === 0) {
+        return null;
```

Comment:
To clarify, does this mean that the overlay will be empty if there were no tags defined?

Reply:
Yes. It'd mean there were no criteria, since every criteria (once loaded into state) should have at least one tag. Should never happen, but better this than some random error, imo.

Reply:
Gotcha, makes sense. Maybe we should log something, too? Makes sense that it should never happen, but I feel like if there was a blank overlay and someone looked at the console and no problems were reported, it would feel very buggy. Although I'm guessing there should be other errors in the console if that were to happen?

Reply:
Added error log